source/auth/auth.rst

```diff
@@ -971,6 +971,12 @@ gssapiServiceName (deprecated)
 	An alias for ``authMechanismProperties=SERVICE_NAME:mongodb``.
 
 
+Errors
+------
+
+Drivers SHOULD raise an error if the ``authSource`` option is specified in the connection string with an empty value, e.g. ``mongodb://localhost/admin?authSource=``.
```

Comment:
Why SHOULD instead of MUST? Is there a concern that being prescriptive here would result in drivers raising errors for existing applications after they upgrade?

I noticed that https://github.com/mongodb/specifications/pull/714 is only addressing integer and boolean types. Any reason to omit string types there? It seems like the appropriate ticket to clarify that `foo=&...` be parsed as an empty string for the `foo` option (assuming it expects a string).

Reply:
In response to my second paragraph, I missed this comment from @jstewart-mongo in Slack:

> I glossed over this before, but the connection string spec contains the following in the section > Key Value Pair: Value (optional) -- the value if provided otherwise it defaults to an empty string.

Point noted on that, but my earlier question about MUST still stands.

Reply:
I did have the concern about the upgrade, but I do believe that an error should be raised, so I have changed the wording to use MUST.

Reply:
> I have changed the wording to use MUST.

The PR still says "SHOULD". Has that change not been pushed?